### PR TITLE
Aiohttp class

### PR DESCRIPTION
**Pull request by exageraldo**

### CODE REVIEW

File: src/connexion/extended/auth_paths_extd.py [hunk 1]
```diff
@@ -23,10 +32,11 @@ def __init__(self, specification, base_path=None, arguments=None,
...
                  ):
-        if auth_all_paths:
-            self._handle_spec = self._mount_sec_operation_view_function
-        else:
-            self._handle_spec = lambda func: func
+
+        self._handle_spec = (
+            self._mount_sec_operation_view_function if auth_all_paths
+            else lambda func: func
+        )
```
Comment (exageraldo) [inline src/connexion/extended/auth_paths_extd.py]: Changed to a ternary operation

File: src/connexion/extended/auth_paths_extd.py [hunk 2]
```diff
@@ -43,16 +53,6 @@ def __init__(self, specification, base_path=None, arguments=None,
...
-    logger = logging.getLogger('connexion.apis.flask_api')
-
-    @property
-    def _handlers(self):
-        # type: () -> InternalHandlers
-        if not hasattr(self, '_internal_handlers'):
-            self._internal_handlers = InternalHandlers(self.base_path, self.options, self.specification)
-        return self._internal_handlers
-
     def _mount_sec_operation_view_function(self, function):
```
Comment (exageraldo) [inline src/connexion/extended/auth_paths_extd.py]: I moved this method from `FlaskApi` to `AbstractAPI`, to reuse it in `AioHttpApi` class.

File: src/connexion/extended/auth_paths_extd.py [hunk 3]
```diff
@@ -98,7 +109,89 @@ def add_openapi_yaml(self):
...
+        AbstractAPI.__init__(self, *args, **kwargs)
+
+        aiohttp_jinja2.setup(
+            self.subapp,
+            loader=jinja2.FileSystemLoader(
+                str(self.options.openapi_console_ui_from_dir)
+            )
+        )
+        middlewares = self.options.as_dict().get('middlewares', [])
+        self.subapp.middlewares.extend(middlewares)
```
Comment (exageraldo) [inline src/connexion/extended/auth_paths_extd.py]: Rewriting the entire initialization method to override the `AbstractAPI` initialization inside.

File: src/connexion/extended/auth_paths_extd.py [hunk 4]
```diff
@@ -98,7 +109,89 @@ def add_openapi_yaml(self):
...
+
+    def _base_path_for_prefix(self, request):
+        """
+        returns a modified basePath which includes the incoming request's
+        path prefix.
+        """
+        if isinstance(request, ConnexionRequest):
+            request = request.context
```
Comment (exageraldo) [inline src/connexion/extended/auth_paths_extd.py]: Some decorators change the object type, so we have to access the right object this way.

File: src/connexion/extended/auth_paths_extd.py [hunk 5]
```diff
@@ -98,7 +109,89 @@ def add_openapi_yaml(self):
...
+
+        openapi_spec_path_yaml = \
+            self.options.openapi_spec_path[:-len("json")] + "yaml"
+        self.logger.debug('Adding spec yaml: %s/%s', self.base_path,
+                     openapi_spec_path_yaml)
+        self.subapp.router.add_route(
+            'GET',
+            openapi_spec_path_yaml,
+            self._handle_spec(self._get_openapi_yaml)
+        )
```
Comment (exageraldo) [inline src/connexion/extended/auth_paths_extd.py]: Same idea that we used in Flask

**Review by exageraldo — COMMENTED** [review]
Just some points

**Review by gahag — APPROVED** [review]
Tchop demais!

File: setup.py [hunk 6]
```diff
@@ -5,6 +5,9 @@
     name="connexion_auth_paths_extd",
     version="0.0.2-dev",
     install_requires=[
-        "connexion~=2.10.0"
-    ]
+        "connexion"
+    ],
+    extras_require={
+        'aiohttp': "connexion[aiohttp]"
+    }
```
Comment (exageraldo) [inline setup.py:12]: I am not sure if this can be an "extra".

File: src/connexion/extended/auth_paths_extd.py [hunk 7]
```diff
@@ -1,11 +1,20 @@
 import logging
 
+from aiohttp import web
+import aiohttp_jinja2
+import jinja2
+from aiohttp.web_exceptions import HTTPPermanentRedirect
+from aiohttp.web_middlewares import normalize_path_middleware
```
Comment (exageraldo) [inline src/connexion/extended/auth_paths_extd.py]: If the aiohttp dependencies are extra, it will get an error when trying to import these packages.

Comment (exageraldo) [reply src/connexion/extended/auth_paths_extd.py]: Is it a good idea to import inside the class?

Comment (exageraldo) [reply src/connexion/extended/auth_paths_extd.py]: Maybe it is better to split it into files anyway.

**Comment by exageraldo** [comment]
Next step:
Split the classes into different files to enable a minimal installation (with no extra) without the aiohttp dependencies.

**Pull request merged**